### ISSUE TITLE
Some safety concerning the maximum allowed predicate arity

### ISSUE THREAD

**Issue by jburse** [SWI-Prolog/swipl-devel#791]
This one crashes on a Mac, didn't try yet on Windows. Also didn't try 
yet in command line, to see whether there is some message.
```
/* SWI-Prolog 8.3.19 */

?- functor(F,f,10000), call(F,a).
%%% crash
```

**Comment by JanWielemaker** [comment]
Thanks.  Should be fixed with bfa78b2ff2d9606507017523e1ddbae2cad9ea29, which also addresses most of the related issues.   Not 100% sure about the above as that doesn't reproduce.   Various related ways to call predicates with too many arguments did crash.

**Comment by JanWielemaker** [comment]
This issue has been mentioned on **SWI-Prolog**. There might be relevant details there:

https://swi-prolog.discourse.group/t/max-arity-status/2464/13
